event/consumer_test.go

```diff
@@ -0,0 +1,113 @@
+package event_test
```

Comment:
The tests use a different package intentionally so they only have access to the public interface of the package.